src/main.rs

```diff
@@ -225,54 +286,119 @@ async fn main() -> anyhow::Result<()> {
 
         let rate = {
             let rates_lock = rates_mutex.lock().unwrap();
-            match compute_average(&*rates_lock) {
-                Some(r) => r,
+            match compute_median(&*rates_lock) {
+                Some(r) => r * &million, /* multiply with 1000000 microCCD/CCD to convert the */
+                // unit to microCCD/Eur
                 None => {
-                    log::error!("Unable to compute average for update");
+                    log::error!("Unable to compute median for update");
                     continue;
                 }
             }
         }; // drop lock
-        log::debug!("Computed average: {:#?}", rate);
+        log::debug!("Computed median: {} microCCD/Eur", rate);
+
+        // Calculates the relative change from the prev_rate, which should be the
+        // current exchange rate on chain, and our proposed update:
+        let diff = relative_change(&prev_rate, &rate);
+        if rate > prev_rate {
+            // Rate has increased
+            if diff > halt_increase_threshold {
+                log::error!(
+                    "New update violates halt threshold, changing from {} to {} is an ~{} % \
+                     increase",
+                    prev_rate,
+                    rate,
+                    diff.round()
+                );
+                bail!("Halt threshold violated");
+            } else if diff > warning_increase_threshold {
+                log::warn!(
+                    "New update violates warning threshold, changing from {} to {} has ~{} % \
+                     increase",
+                    prev_rate,
+                    rate,
+                    diff.round()
+                );
+                stats.increment_warning_threshold_violations();
+            }
+        } else {
+            // Rate has decreased
+            if diff > halt_decrease_threshold {
```

Comment:
No, I think this is correct, the difference is positive. 

But maybe I should change it so that relative_change returns a negative number, because currently it calculates the absolute relative change.

Reply:
No I think this is fine. I just got confused :)